pandas/tests/indexes/test_base.py

```diff
@@ -1557,6 +1557,11 @@ def test_string_index_repr(self):
 
                 self.assertEqual(coerce(idx), expected)
 
+    def test_map(self):
+        df = pd.DataFrame([[0, 1], [2, 3]], columns=['c1', 'c2'], index=['i1', 'i2'])
+        df.index = df.index.map(lambda x: x.upper())
+        self.assertTrue(df.index.equals(Index(['I1', 'I2'])))
```

Comment:
Pls add following cases:

- dtype changes
- has ``name`` (and ``map`` preserves the ``name``)